src/components/swap/SwapRoute.tsx

```diff
@@ -1,64 +1,83 @@
 import { Trans } from '@lingui/macro'
-import { useWeb3React } from '@web3-react/core'
 import Column from 'components/Column'
-import { LoadingRows } from 'components/Loader/styled'
 import RoutingDiagram from 'components/RoutingDiagram/RoutingDiagram'
+import { RowBetween } from 'components/Row'
 import { SUPPORTED_GAS_ESTIMATE_CHAIN_IDS } from 'constants/chains'
 import useAutoRouterSupported from 'hooks/useAutoRouterSupported'
-import { ClassicTrade } from 'state/routing/types'
+import { ClassicTrade, SubmittableTrade } from 'state/routing/types'
+import { isClassicTrade } from 'state/routing/utils'
 import { Separator, ThemedText } from 'theme/components'
+import { useFormatter } from 'utils/formatNumbers'
 import getRoutingDiagramEntries from 'utils/getRoutingDiagramEntries'
 
 import RouterLabel from '../RouterLabel'
+import { UniswapXDescription } from './GasBreakdownTooltip'
 
-export default function SwapRoute({ trade, syncing }: { trade: ClassicTrade; syncing: boolean }) {
-  const { chainId } = useWeb3React()
-  const autoRouterSupported = useAutoRouterSupported()
+// TODO(WEB-2022)
+// Can `trade.gasUseEstimateUSD` be defined when `chainId` is not in `SUPPORTED_GAS_ESTIMATE_CHAIN_IDS`?
+function getGasPrice(trade: ClassicTrade) {
+  const chainId = trade.inputAmount.currency.chainId
+  if (!trade.gasUseEstimateUSD || !SUPPORTED_GAS_ESTIMATE_CHAIN_IDS.includes(chainId)) return undefined
 
-  const routes = getRoutingDiagramEntries(trade)
+  return trade.gasUseEstimateUSD === 0 ? '<$0.01' : '$' + trade.gasUseEstimateUSD.toFixed(2)
```

Comment:
you should just return a number here and let a formatter do the formatting after its returned